templates/vm.xml.j2

```diff
@@ -1,38 +1,45 @@
 <domain type='{{ libvirt_vm_engine }}'>
-  <name>{{ libvirt_vm_name }}</name>
-  <memory>{{ libvirt_vm_memory_mb | int * 1024 }}</memory>
-  <vcpu>{{ libvirt_vm_vcpus }}</vcpu>
+  <name>{{ vm.name }}</name>
+  <memory>{{ vm.memory_mb | int * 1024 }}</memory>
+  <vcpu>{{ vm.vcpus }}</vcpu>
   <clock sync="localtime"/>
+  <on_poweroff>destroy</on_poweroff>
+  <on_reboot>restart</on_reboot>
+  <on_crash>destroy</on_crash>
   <os>
-    <type arch='{{ libvirt_vm_arch }}'{% if libvirt_vm_machine is not none %} machine='{{ libvirt_vm_machine }}'{% endif %}>hvm</type>
+    <type arch='{{ libvirt_vm_arch }}'{% if vm.machine is defined %} machine='{{ vm.machine | default('pc-1.0') }}'{% endif %}>hvm</type>
```

Comment:
Not sure the logic is quite right here. Docs say:

Default is `None` if `engine` is `kvm`, otherwise `pc-1.0`.

You can use {% set var=val %} if the logic gets hairy.